collect_app/src/main/java/org/odk/collect/android/activities/FormMapActivity.java

```diff
@@ -75,9 +113,16 @@
         TextView titleView = findViewById(R.id.form_title);
         titleView.setText(viewModel.getFormTitle());
 
-        Context context = getApplicationContext();
-        MapProvider.createMapFragment(context)
-            .addTo(this, R.id.map_container, this::initMap, this::finish);
+        if (map == null) { // tests set their maps directly
+            Context context = getApplicationContext();
+            map = MapProvider.createMapFragment(context);
+        }
+
+        if (map != null) {
+            map.addTo(this, R.id.map_container, this::initMap, this::finish);
+        } else {
+            finish(); // The configured map provider is not available
```

Comment:
I haven't tested/tried it yet but I think this will address @mmarciniak90's find at https://github.com/opendatakit/collect/pull/3503#issuecomment-567511721.